controller/server/gui.ml

```diff
@@ -263,57 +261,125 @@ module NetworkGui = struct
       >|= List.filter (fun s -> not internet_connected || s |> Service.is_connected)
     in
 
+    let blur_service_proxy_password s =
+      let open Service in
+      let blur p = Proxy.validate p |> Option.map (Proxy.to_string ~hide_password:true) in
+      { s with proxy = s.proxy |> Base.Fn.flip Option.bind blur }
+    in
+
+    let after_restart_proxy =
+      services
+      |> List.find_map (fun s ->
+          if s.Service.autoconnect then
+            Option.bind s.Service.proxy Proxy.validate
+          else
+            None)
+    in
+
+    let proxy_infos =
+      [ proxy |> Option.map (Proxy.to_string ~hide_password:true)
+      ; (if after_restart_proxy <> proxy then
```

Comment:
The proxy detection must be different between startup (done by the Python script) and during the request. In both cases, it’s looking for a service which has _AutoConnect_ set to _true_, and pick the 1st one.

Using ethernet, but no wifi, I don’t have the problem. Do you have both ethernet and wifi at the same time ?

 Can you reproduce this directly on the controller ? Or in the qemu VM ?

Reply:
> Using ethernet, but no wifi, I don’t have the problem. Do you have both ethernet and wifi at the same time ?

Only Ethernet is connected.

> Can you reproduce this directly on the controller ? Or in the qemu VM ?

I get this result when testing in a full PlayOS system installed on real hardware. I see `playos-controller: [INFO] proxy: none` in the logs.

Is it possible that the detection initially finds no proxy, since it runs before the Ethernet interface appears or the router on the other side activates?

Reply:
I am reproducing the problem on my system, I got insight by adding logs to the detection script:
https://github.com/guyonvarch/playos/commit/fbc46aed937455b5dac82a8530d7625d940531b1

Reply:
Since we are now configuring the proxy inside the running apps and no longer using environment variables, might it be an idea to subscribe to connection changes via DBus and update proxy information accordingly?

Reply:
Yes, that would be less brittle, and it would be more valuable to users.

I just verified that we can change the proxy on the fly on the kiosk, it’s working well. So, I think there is nothing preventing us to set this up. I’ll have a try at this. :slightly_smiling_face:

Reply:
Live reload of the proxy in the kiosk: https://github.com/dividat/playos/pull/54/commits/41a69d4257ca5ae0b4ead11ec0e3a5602f67d626

I wrote about an unlikely, but still potential issue on initialization, I’ll think how to take care of it.

But I’ll take care of the controller first. My idea is not to listen to _connman_ for changes in the controller, but rather to get a fresh proxy everytime we need it.

Reply:
Nice, did a quick test of the current state and it seems to perform well!

> I wrote about an unlikely, but still potential issue on initialization, I’ll think how to take care of it.

Can the problem be avoided by actively polling once for the current state just after setting up the monitoring?

Reply:
I am having an issue on _qemu VM_: if a select a non existent proxy, the network page stills in the controller stills indicate I am connected. I’ll see what’s happening. I don’t have the problem running the controller directly on my system.

> Can the problem be avoided by actively polling once for the current state just after setting up the monitoring?

￼Yes, that should do it. I am not completely sure at which point we can be sure the monitoring is set up. I’ll have to understand better how the dbus main loop is working.

Reply:
> I am having an issue on qemu VM: if a select a non existent proxy, the network page in the controller stills indicate I am connected. I’ll see what’s happening. I don’t have the problem running the controller directly on my system.

Fixed with https://github.com/dividat/playos/pull/54/commits/baa96d0a4c0edc699809d3975f83b87f70eae779, which is recomputing the internet status on the fly, instead of using a computed state.

Reply:
> Can the problem be avoided by actively polling once for the current state just after setting up the monitoring?

Initializing the proxy after the monitoring is on: https://github.com/dividat/playos/pull/54/commits/6587316e832854245885c31bb2706cd3eacb2ef0

I can’t say if the monitoring is on directly after the call is made, but at least, the loop behind has nothing to do with it, I verified it by adding a sleep before the loop is call, and verifying proxy updates are detected and applied.